model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/DesignElementHandle.java

```diff
@@ -705,14 +703,13 @@ public List getMethods() {
 	 *                           value is invalid.
 	 * @see #setProperty(String, Object)
 	 */
-
-	public void setProperties(Map properties) throws SemanticException {
+	public void setProperties(Map<?, ?> properties) throws SemanticException {
 		if (properties == null) {
 			return;
 		}
 
-		for (Iterator iter = properties.entrySet().iterator(); iter.hasNext();) {
-			Entry entry = (Entry) iter.next();
+		for (Iterator<?> iter = properties.entrySet().iterator(); iter.hasNext();) {
```

Comment:
A for-each loop would be much less verbose.  Also, the entrySet is already of type `Entry<?,?>` so this use of an iterator of type `<?>`  loses the type and then you need a cast below.

```
for (Map.Entry<?, ?> entry : properties.entrySet()) {
}
```
```

Reply:
I didn't understand what your suggestion is. The most suggestions are comming from the eclipse-java-side as solving hint.

Reply:
Here is the full replacement:
```
	public void setProperties(Map<?, ?> properties) throws SemanticException {
		if (properties == null) {
			return;
		}

		for (Entry<?, ?> entry : properties.entrySet()) {
			setProperty((String) entry.getKey(), entry.getValue());
		}
	}
```
It's extremely rare that you should need to use an Iterator anywhere, i.e., only if you need to call remove on it.

Reply:
Change is done.